azure-kusto-ingest/azure/kusto/ingest/_streaming_ingest_client.py

```diff
@@ -88,27 +86,29 @@ def ingest_from_stream(self, stream_descriptor, ingestion_properties):
         else:
             stream = stream_descriptor.stream
 
-        if stream_descriptor.size >= self._streaming_ingestion_size_limit:
-            raise KustoStreamMaxSizeExceededError()
-
         if (
             ingestion_properties.format in self._mapping_required_formats
             and ingestion_properties.mapping_reference is None
         ):
             raise KustoMissingMappingReferenceError()
 
-        client_request_properties = ClientRequestProperties()
-        client_request_properties.set_option("content_length", str(stream_descriptor.size))
-        if stream_descriptor.source_id:
-            client_request_properties.set_option("x-ms-client-request-id", str(stream_descriptor.source_id))
-        if stream_descriptor.is_zipped_stream:
-            client_request_properties.set_option("Content-Encoding", "gzip")
+        if not stream_descriptor.is_compressed:
+            zipped_stream = BytesIO()
+            buffer = stream.read()
+            with GzipFile(filename="data", fileobj=zipped_stream, mode="wb") as f_out:
+                if isinstance(buffer, string_types):
+                    data = bytes(buffer) if PY2 else bytes(buffer, "utf-8")
+                    f_out.write(data)
+                else:
+                    f_out.write(buffer)
+            zipped_stream.seek(0)
+            stream = zipped_stream
 
         self._kusto_client.execute_streaming_ingest(
             ingestion_properties.database,
             ingestion_properties.table,
             stream,
             ingestion_properties.format.name,
-            client_request_properties,
+            None,
```

Comment:
I try to avoid passing `None`s, if you have position mismatch, using named params (also smart for when params aren't inferred from name, for example: `get_something(param)` - kidna obvious what param is , but `get_something(p1,p2,p3)` - is messy, much easier like `get_something(db=p1,table=p2)` ) .

Reply:
Neat =)